go/ct/rlz/conditions.go

```diff
@@ -719,3 +719,98 @@ func (c *hasNotSelfDestructed) GetTestValues() []TestValue {
 func (c *hasNotSelfDestructed) String() string {
 	return "hasNotSelfDestructed()"
 }
+
+////////////////////////////////////////////////////////////
+// In Range 256 From Current Block
+
+type inRange256FromCurrentBlock struct {
+	blockNumber BindableExpression[U256]
+}
+
+func InRange256FromCurrentBlock(blockNumber BindableExpression[U256]) Condition {
+	return &inRange256FromCurrentBlock{blockNumber}
+}
+
+func (c *inRange256FromCurrentBlock) Check(s *st.State) (bool, error) {
+	paramBlockNumber, err := c.blockNumber.Eval(s)
+	if !paramBlockNumber.IsUint64() {
+		return false, nil
+	}
+	uintParam := paramBlockNumber.Uint64()
+	if err != nil {
+		return false, err
+	}
+	bottom := uint64(0)
+	if s.BlockContext.BlockNumber > 256 {
+		bottom = s.BlockContext.BlockNumber - 256
+	}
+
+	return bottom <= uintParam && uintParam < s.BlockContext.BlockNumber, nil
+}
+
+func (c *inRange256FromCurrentBlock) Restrict(generator *gen.StateGenerator) {
+	paramVariable := c.blockNumber.GetVariable()
+	c.blockNumber.BindTo(generator)
+	generator.AddBlockNumberOffset(paramVariable, 1)
+}
+
+func (c *inRange256FromCurrentBlock) GetTestValues() []TestValue {
+	property := Property(c.String())
+	domain := BlockNumberRangeDomain{}
+	restrict := func(generator *gen.StateGenerator, offset int64) {
+		paramVariable := c.blockNumber.GetVariable()
+		c.blockNumber.BindTo(generator)
+		generator.AddBlockNumberOffset(paramVariable, offset)
+	}
+	testValues := []TestValue{}
+	var blockNumberDomain BlockNumberRangeDomain
+	for _, value := range blockNumberDomain.Samples(0) {
+		testValues = append(testValues, NewTestValue(property, domain, value, restrict))
+	}
+	return testValues
+}
+
+func (c *inRange256FromCurrentBlock) String() string {
+	return fmt.Sprintf("blockRangeOffset(%v)", c.blockNumber)
+}
+
+////////////////////////////////////////////////////////////
+// Out Of Range 256 From Current Block
+
+type outOfRange256FromCurrentBlock struct {
+	blockNumber BindableExpression[U256]
+}
+
+func OutOfRange256FromCurrentBlock(blockNumber BindableExpression[U256]) Condition {
+	return &outOfRange256FromCurrentBlock{blockNumber}
+}
+
+func (c *outOfRange256FromCurrentBlock) Check(s *st.State) (bool, error) {
+	// no check for overflow because it is considered outside the 256 range
+	paramBlockNumber, err := c.blockNumber.Eval(s)
+	if err != nil {
+		return false, err
+	}
+	bottom := uint64(0)
+	if s.BlockContext.BlockNumber > 256 {
+		bottom = s.BlockContext.BlockNumber - 256
+	}
+	uintParam := paramBlockNumber.Uint64()
+	return uintParam >= s.BlockContext.BlockNumber || bottom > uintParam, nil
```

Comment:
This should be simplified by reusing `inRange256FromCurrentBlock`'s implementation of the `Check` function:
```
res, err := InRange256FromCurrentBlock(c.blockNumber).Check(s)
return !res, err
```